tools/hidl/hidlcomp.cpp

```diff
@@ -5757,26 +5792,59 @@ void EspServInfo::write_esp_binding()
             serviceCacheGroupID.set(name_);
         outf("\tsetCacheGroupID(nullptr, \"%s\");\n", serviceCacheGroupID.str());
     }
-
     outs("}\n");
 
     //
     // Create init_metrics for execution profiling
-    outf("\nvoid C%sSoapBinding::init_metrics()\n", name_);
-    outs("{\n");
-
-    outf("#ifdef ESP_SERVICE\n");
-
-    // For each method with execution profiling enabled, add code to initialize the histogram metric
-    for (mthi=methods; mthi!=NULL; mthi=mthi->next)
+    if (executionProfilingEnabled)
     {
-        if (mthi->isExecutionProfilingEnabled())
-        {
-            outf("\t%s = registerProfilingMetric(\"%s\", \"\", \"%s\");\n", mthi->getExecutionProfilingMetricName(), mthi->getName(), mthi->getExecutionProfilingOptions().c_str());
+        outf("#ifdef ESP_SERVICE_%s\n", name_);
+        outf("\nvoid C%sSoapBinding::init_metrics()\n", name_);
+        outs("{\n");
+        outs("\tstd::string processName(queryProcessName());\n");  // process/app name
+        outs("\tstd::string metricName;\n");
+        outs("\tmetricName.reserve(100);\n");
```

Comment:
I don't think this reserve buys you much?  get_metric_name()  does multiple appends to a locally declared string and then returns it by value... the return might be optimized by the compiler?   But I think if you did the reserve and then passed the string by reference into get_metric_name() then it might be more efficient doing those multiple appends directly into this variable with the length reserved?

Reply:
Correct. Before ensuring all generated code was ifdef'd out, I had an empty method, so I needed to return something, hence the local variable. Now that no code is created if not instrumented, I changed it to pass a reference as indicated.